lib/navigation.php

```diff
@@ -435,3 +435,80 @@ function gutenberg_register_navigation_post_type() {
 	register_post_type( 'wp_navigation', $args );
 }
 add_action( 'init', 'gutenberg_register_navigation_post_type' );
+
+/**
+ * Disable block editor for wp_navigation type posts so they can be managed via the UI.
+ *
+ * @param bool   $value Whether the CPT supports block editor or not.
+ * @param string $post_type Post type.
+ *
+ * @return bool
+ */
+function gutenberg_disable_block_editor_for_navigation_post_type( $value, $post_type ) {
+	if ( 'wp_navigation' === $post_type ) {
+		return false;
+	}
+
+	return $value;
+}
+
+add_filter( 'use_block_editor_for_post_type', 'gutenberg_disable_block_editor_for_navigation_post_type', 10, 2 );
+
+/**
+ * This callback disables content editor for wp_navigation type posts.
+ * Content editor cannot handle wp_navigation type posts correctly.
+ *
+ * @param WP_Post $post An instance of WP_Post class.
+ */
+function gutenberg_disable_content_editor_for_navigation_post_type( $post ) {
+	$post_type = get_post_type( $post );
+	if ( 'wp_navigation' !== $post_type ) {
+		return;
+	}
+
+	remove_post_type_support( $post_type, 'editor' );
```

Comment:
So this has to hapen b/c if the post type definition itself does not support `editor` we get into the rest problem not allowing saving at all. That would be good to be added to the docblock of the function.